lib/User_LDAP.php

```diff
@@ -152,6 +152,13 @@ public function checkPassword($uid, $password) {
 		if ($this->config->getSystemValue('enable_avatars', true) === true) {
 			$this->userManager->registerAvatarHook($userEntry);
 		}
+
+		// hack to store a samaccountname in the session
+		$samaccountname = $userEntry->getSAMAccountName();
+		if ($samaccountname !== null) {
+			\OC::$server->getSession()->set('samaccountname', $samaccountname);
```

Comment:
I'd like to call this "alternateLogin" or something more generic to make it reusable

@DeepDiver1975 are you ok with this approach ?
also see https://github.com/owncloud/windows_network_drive/pull/175 and https://github.com/owncloud/core/pull/31951 for usage

if we agree I'll adjust the PRs to rename the attributes

Reply:
had a chat with @DeepDiver1975 and he's ok with this